### ISSUE TITLE
Unique Symbol resolution with JSDoc

### ISSUE THREAD

**Issue by 8HoLoN** [microsoft/TypeScript#56757]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.85.0
- OS Version: Windows_NT x64 10.0.19045

Steps to Reproduce:
1.  paste in a .js file the following JSDoc/Typescript definition
```js
/**
 * @typedef {unique symbol} Foo
 */

/**
 * @typedef { {
*  [key: Foo]: boolean;
* } } A
*/
```
2. When the type A is hover, the intellisense show
```
type A = {
    [key: symbol]: boolean;
}
``` 
The allowed key should be only Foo symbol, not any symbol provided. The type 'unique symbol' seems not to be taken in account.

**Comment by a-tarasyuk** [comment]
@sandersn Is it allowed to use `{unique symbol}` as type alias?

**Comment by sandersn** [comment]
No, `type Foo = unique symbol` is an error in Typescript.

**Comment by sandersn** [comment]
I think that means the correct fix here is to add the same error to `@typedef` in JS.

**Comment by 8HoLoN** [comment]
@sandersn so what is the correct syntax for the desired behavior ?

**Comment by sandersn** [comment]
This is how I'd write it in Typescript:

```ts
const s: unique symbol = Symbol()
type A = { [s]: boolean }
```

So I guess the equivalent JS should be

```js
/** @type {unique symbol} */
const s = Symbol()
/** @typedef {{ [s]: boolean }} A */
```

And this in fact *also* fails to work in JS, with an incorrect error that 's' must have type unique symbol.

**Comment by 8HoLoN** [comment]
Ok I understand the type should be declared directly with the associated symbol declaration. Because using type alias for symbol would allow multiple reference for that type which is not possible per definition.

Not ok:
```js
/**
* @typedef {unique symbol} Foo
*/

/**
* @type {Foo}
*/
const foo = Symbol()
```

Ok:
```js
/**
* @type {unique symbol}
*/
const foo = Symbol()
```

Right?

**Comment by sandersn** [comment]
Here's a full repro for the issue I found:

```js
/** @type {unique symbol} */
const s = Symbol()
/** @typedef {{ [s]: boolean }} A - should be OK in JS */
/** @type {unique symbol} */
const v = Symbol()
/** @typedef {{ [key: v]: boolean }} D - should error in JS */
```

compare to TS

```ts
const t: unique symbol = Symbol()
type B = { [t]: boolean } // should be OK in TS (and is)
const u: unique symbol = Symbol()
type C = { [key: u]: boolean } // should error in TS (and does)
```

The original code works identically between TS and JS and is working as intended. I filed #56800 to track this bug.

**Comment by sandersn** [comment]
@8HoLoN Just saw your latest comment. Yes, that's right. Except that's **also** broken. =( I filed #56800 to track it.

**Comment by 8HoLoN** [comment]
I have another related question, how to write it if the symbol is declare in an object (not as a independant variable):
This is not working: 
```ts
const v: { t: unique symbol } = {
    t: Symbol()
}

const f = (a: typeof v.t) => {
    a
}
```

**Comment by sandersn** [comment]
In a .ts file, the compiler says you are missing a `readonly` in the type: `const v: { readonly t: unique symbol } = ...`

If you add that, then `f(v.t)` works, which is I think what you want.
However, there's an error on `t: Symbol()` which I can't get rid of. I would write this like so:

```ts
const t: unique symbol = Symbol()
const v: {t : type of t } = { t }
const f = (a: typeof v.t) => a
f(v.t)
```
